### PR TITLE
BAU: Fix broken apt packages because /tmp permissions. Port nginx com…

### PR DESCRIPTION
**Pull request by carinadigital**
…pilation to Dockerfile.

### CODE REVIEW

File: opg-nginx-1604/Dockerfile [hunk 1]
```diff
@@ -1,16 +1,57 @@
...
     mkdir -p /app/public /etc/nginx/app.conf.d                   &&\
     echo "Hello World from opguk/nginx" > /app/public/index.html
+WORKDIR /
+
+RUN echo "clean up " &&\
+    rm /tmp/nginx-${NGINX_VERSION}.tar.gz &&\
+    rm -rf /tmp/nginx-* &&\
+    apt-get remove -y --purge build-essential &&\
+    apt clean && apt -y autoremove && \
+    rm -rf /var/lib/cache/* /var/lib/log/* /tmp/* /var/tmp/*
```
Comment (damacus) [inline opg-nginx-1604/Dockerfile]: These steps should be in the previous `RUN` step. 

I just tried it and it reduces the image size from `741MB` to `594MB`

File: opg-nginx-1604/Dockerfile [hunk 2]
```diff
@@ -1,16 +1,57 @@
...
+RUN echo "Setup nginx user and dirs" &&\
+    useradd -r nginx &&\
+    mkdir -p /var/log/nginx &&\
+    mkdir -p /var/lib/nginx/body &&\
+    chown nginx:nginx /var/log/nginx
 
-RUN echo "Compilling nginx"                                      &&\
-    . /tmp/compile_nginx.sh                                      &&\
+WORKDIR /tmp
+RUN echo "Compiling nginx" &&\
```
Comment (damacus) [inline opg-nginx-1604/Dockerfile]: Can you align these for me please?

Comment (carinadigital) [reply opg-nginx-1604/Dockerfile]: What alignment system do you want? It' not clear

Comment (damacus) [reply opg-nginx-1604/Dockerfile]: it's easier to read when `&&\` are all aligned.

**Comment by damacus** [comment]
This kind of alignment https://gist.github.com/damacus/0e730f0475173b2ebaa2cc34060412c9

**Comment by damacus** [comment]
@dragoonis was quickly able to spot an extra `&&\` in the code when it was aligned as per the example.

**Pull request merged**